apps/src/templates/census2017/CensusMapReplacement.jsx

```diff
@@ -0,0 +1,402 @@
+  draggableOption = false;
+  resizeThrottleTimeoutMs = 500;
+  maxHeightPercentage = 0.65;
+
+  componentWillUnmount() {
+    if (this.popup) {
+      this.popup.remove();
+    }
+  }
+
+  componentDidMount = () => {
+    this.initializeMap();
+
+    $(window).resize(() => {
+      // Throttle calling of resizeMap
+      clearTimeout(this.resizeThrottleTimerId);
+      this.resizeThrottleTimerId = setTimeout(
+        this.resizeMap,
+        this.resizeThrottleTimeoutMs
+      );
+    });
+  };
+
+  displayPopup(content, loc) {
+    if (this.popup) {
+      this.popup.remove();
+    }
+    this.popup = new mapboxgl.Popup()
+      .setLngLat(loc)
+      .setDOMContent(content)
+      .addTo(this.map);
+  }
+
+  formatCoordinatesString(latitude, longitude) {
+    return longitude + ',' + latitude;
+  }
+
+  onPointClick(_this, e) {
+    var coordinates = e.features[0].geometry.coordinates.slice();
+    var properties = e.features[0].properties;
+
+    while (Math.abs(e.lngLat.lng - coordinates[0]) > 180) {
+      coordinates[0] += e.lngLat.lng > coordinates[0] ? 360 : -360;
+    }
+
+    var content = _this.generateInfoWindowElement(
+      properties.school_id,
+      properties.school_name,
+      properties.school_city,
+      properties.school_state,
+      _this.formatCoordinatesString(coordinates[0], coordinates[1]),
+      properties.teaches_cs ? properties.teaches_cs : ''
+    );
+
+    _this.displayPopup(content, coordinates);
+  }
+
+  initializeMap = () => {
+    this.map = new mapboxgl.Map({
+      container: 'mapbox',
+      style: 'mapbox://styles/codeorg/cjyudafoo004w1cnpaeq8a0lz',
+      zoom: 3,
+      minZoom: 1,
+      center: [-98, 39]
+    });
+    var _this = this;
+
+    this.map.on('load', function() {
+      _this.map.addSource('censustiles', {
+        type: 'vector',
+        url: 'mapbox://codeorg.censustiles'
+      });
+
+      _this.map.addLayer({
+        id: 'census-schools',
+        type: 'circle',
+        source: 'censustiles',
+        'source-layer': 'census',
+        layout: {
+          visibility: 'visible'
+        },
+        paint: {
+          'circle-radius': 4,
+          'circle-color': [
+            'match',
+            ['get', 'teaches_cs'],
+            ['NO', 'HISTORICAL_NO'],
+            '#989CF8',
+            ['MAYBE', 'HISTORICAL_MAYBE'],
+            '#FFFDA6',
+            'white'
+          ],
+          'circle-stroke-width': 1,
+          'circle-stroke-color': '#000000'
+        },
+        filter: [
+          'any',
+          ['!=', 'teaches_cs', 'YES'],
+          ['!=', 'teaches_cs', 'HISTORICAL_YES']
+        ]
+      });
+      _this.map.addLayer({
+        id: 'census-schools-teaching-cs',
+        type: 'symbol',
+        source: 'censustiles',
+        'source-layer': 'census',
+        layout: {
+          'icon-image': 'marker-15-green',
+          'icon-allow-overlap': true,
+          'icon-size': ['interpolate', ['linear'], ['zoom'], 1, 0.7, 14, 1.6]
```

Comment:
Did we choose these numbers, and are they meaningful in some way specific to us?

Reply:
I chose them. Just added a comment explaining what they do, though the numbers themselves are just based on what looked alright to me.